numba_cuda/numba/cuda/tests/cudadrv/test_module_init.py

```diff
@@ -0,0 +1,57 @@
+import numpy as np
+
+from numba import cuda
+from numba.cuda.cudadrv.linkable_code import CUSource
+from numba.cuda.testing import CUDATestCase
+
+from cuda.bindings.driver import cuModuleGetGlobal, cuMemcpyHtoD
+
+
+class TestModuleInitCallback(CUDATestCase):
+
+    def setUp(self):
+        super().setUp()
+
+        module = """
+__device__ int num = 0;
+extern "C"
+__device__ int get_num(int &retval) {
+    retval = num;
+    return 0;
+}
+"""
+
+        def set_fourty_two(mod):
+            # Initialize 42 to global variable `num`
+            res, dptr, size = cuModuleGetGlobal(
+                mod.handle.value, "num".encode()
+            )
```

Comment:
Right now we assumes `mod` to be a `ctypesmodule`. But this is about to be changed to `ObjectCode` in #133 .

Reply:
Synced offline - we are still going to proceed assuming we use ctypes module for now.